### ISSUE TITLE
[Feature] Load playwright.config.js only once per run

### ISSUE THREAD

**Issue by truefusion** [microsoft/playwright#7529]
### Feature request

Currently, the config is loaded more than once per run. This makes it problematic for me when I wish to perform certain, specific tasks within the configuration that I only wish to be ran once (e.g. prompting for input). If the configuration would be loaded only once, it would resolve my issue.

**Comment by mxschmitt** [comment]
What are you trying to do before the runs? There are multiple ways of doing something like that:

a) Create a worker fixture with `auto: true`. This gets then triggered one time per worker, see here:
- https://playwright.dev/docs/test-fixtures#worker-fixtures
b) You can leverage the globalSetup functionality, to run something before all your tests will run. This is a one-time execution, see here:
- https://playwright.dev/docs/test-advanced#global-setup-and-teardown

**Comment by truefusion** [comment]
My tests require basic HTTP authentication. Rather than resorting to storing passwords locally, I attempted to implement prompting for input in the config. This works fine the first time around, but playwright seems to require loading the config multiple times. From the looks of it, it doesn't seem like your suggestions will work in my case.

**Comment by mxschmitt** [comment]
Do I understand that correctly that you need to pass the username/email + password from externally into your tests?

Then I would recommend to use environment variables for that which you can read inside your tests using e.g. `process.env.USERNAME`

**Comment by truefusion** [comment]
Are you saying that the `globalSetup` file will be executed before playwright attempts to import from `playwright.config.js` and that whatever I store in `process.env` will be made available to `playwright.config.js`?

**Comment by mxschmitt** [comment]
No, since you specify the path to your globalSetup in the playwright.config.js. Are the values which you want to pass over dynamic and if so how do you decide which values to use?

If its not dynamic, we recommend environment variables, and if dynamic a global setup file which includes the logic to determine the values and sets then the values onto the process.env object so the the tests can access it.

**Comment by dgozman** [comment]
@truefusion I don't think we'll be able to only load config once, so you'll need a workaround.

I think the easiest one would be `globalSetup` that prompts the user and stores the result in `process.env`, as @mxschmitt suggests. I guess you want that available in `playwright.config.js` to set something like `httpCredentials`? You can do the same by overriding the `httpCredentials` fixture by extending from `test`, and using that `test` in your test files:

```js
const base = require('@playwright/test');
const test = base.test.extend({
  httpCredentials: async ({}, use) => {
    await use({
      username: process.env.USERNAME_FROM_GLOBAL_SETUP,
      password: process.env.PASSWORD_FROM_GLOBAL_SETUP
    });
  },
});
module.exports = { ...base, test };
```

**Comment by truefusion** [comment]
I managed to sort something out with `playwright.config.js`  and `process.env`. Extending the base test was not desirable because I end up losing functionality (e.g. `test.describe`), and I am not aware of how to extend the base test in a way that maintains desired functionality. 

Thanks for help!

**Comment by dgozman** [comment]
> Extending the base test was not desirable because I end up losing functionality (e.g. `test.describe`)

I would be very interested in the bug report about lost functionality, because extended `test` should retain all the features by design.

**Comment by mxschmitt** [comment]
I was able to repro. I faced the same issue a few weeks ago and created a new issue to track it: #7552.

**Comment by dgozman** [comment]
@mxschmitt I only see a repro that shows we read test file multiple times, and that's expected. Seems unrelated to `test.describe` not working.

**Comment by truefusion** [comment]
Seems I may have done something wrong the first time around. I am unable to reproduce the issue. I don't know what I may have done to cause the error `describe is not a function.` If I am ever able to reproduce it, I'll let you guys know.
